File: Graph/ScriptTemplate/RadianToDeg/CGRadianToDeg.lua

local CGRadianToDeg = CGRadianToDeg or {}
CGRadianToDeg.__index = CGRadianToDeg

function CGRadianToDeg.new()
    local self = setmetatable({}, CGRadianToDeg)
    self.inputs = {}
    self.outputs = {}
    self.nexts = {}
    return self
end

function CGRadianToDeg:setInput(index, func)
    self.inputs[index] = func
end

function CGRadianToDeg:getOutput(index)
    if self.inputs[0]() then
        return math.deg(self.inputs[0]())
    end
    return nil
end

return CGRadianToDeg

File: Graph/ScriptTemplate/RadianToDeg/CGRadianToDeg.js
/**
 * @file CGRadianToDeg.js
 * @author xuyuan
 * @date 2021/8/28
 * @brief CGRadianToDeg.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const Amaz = effect.Amaz;
const {BaseNode} = require('./BaseNode');

class CGRadianToDeg extends BaseNode {
  constructor() {
    super();
  }

  getOutput(index) {
    return this.inputs[0]() * (180 / Math.PI);
  }
}

exports.CGRadianToDeg = CGRadianToDeg;
